### DISCUSSION TITLE
Open in browser or open path for Android

### DISCUSSION THREAD

**Discussion by jimchen5209** [ppy/osu#22401] [Q&A]
Actions like "view in browser" on beatmap info page, "Export skin" in settings, clicking on "Screenshot saved" notification or clicking "Update available" notification does not do anything on Android, which is expected to open web browser or file explorer.

https://user-images.githubusercontent.com/10269287/214478803-4c6c0ce3-2e05-4221-a6ee-942dc62d751e.mp4

https://user-images.githubusercontent.com/10269287/214479038-8322dbb5-96a4-40bb-80f3-cca8f5819fd0.mp4

Tested devices:
- Samsung Galaxy A71 4G (SM-A715F) Android 13 Build:A715FZHU8DVK1 Default Browser: Google Chrome
- Samsung Galaxy Tab S8 (SM-X700)  Android 13 Build:X700XXU3BVL2  Default Browser: Google Chrome

**Comment by peppy** [comment]
That's weird, this is implemented at a framework level for URLs (but not files). So opening a URL in the browser should *definitely* work.

https://github.com/ppy/osu-framework/blob/9a50bc4b11def8aecff3f6eb0aa49cfd5c1278dd/osu.Framework.Android/AndroidGameHost.cs#L77-L89

Can anyone else confirm this?

**Reply by Theighlin** [reply]
Realme X50 Pro on Android 12 here. I can open the URLs on an external browser.

https://user-images.githubusercontent.com/53081750/214583707-22a383d1-b660-43ac-b9c4-f7f7bce4ec5a.mp4

From the video in the opening post it doesn't look like a firm click, more like dragging. It has to be a tap, and the hitbox of the button itself is very small (i believe my ui scale is higher than 1 here)

**Reply by jimchen5209** [reply]
No, even I "click" it with touch or mouse it just do nothing.

**Reply by jimchen5209** [reply]
Interesting...I just dragged a little bit on the button and it still opens browser after releasing the button on PC.

https://user-images.githubusercontent.com/10269287/214586850-bd589f71-e244-4c49-aaec-946467e5fbbb.mp4
